File: P02 - Kelas Objek/Java/Bag. kedua/MesinCuci.java
// package 

import java.util.Scanner;

public class MesinCuci {
    public String merk;
    public String tipe;
    public int kapasitas;
    public double beratCucian;
    public String modePencucian;
    public boolean airHangat;
    public String teksAir;

    public MesinCuci(String merk, String tipe, int kapasitas, double beratCucian, String modePencucian, boolean airHangat) {
        this.merk = merk;
        this.tipe = (tipe.equalsIgnoreCase("top loading") || tipe.equalsIgnoreCase("front loading")) ? tipe : "top loading";
        this.kapasitas = (kapasitas > 0) ? kapasitas : 1; 
        this.beratCucian = (beratCucian > 0) ? beratCucian : 1; 
        this.modePencucian = (modePencucian.equalsIgnoreCase("ringan") || modePencucian.equalsIgnoreCase("sedang") || modePencucian.equalsIgnoreCase("berat")) ? modePencucian : "sedang";
        this.airHangat = airHangat;
        this.teksAir = (airHangat) ? "menggunakan air hangat" : "tidak menggunakan air hangat";
    }

    public String mencuci() {
        return "Sedang mencuci " + beratCucian + " kg pakaian, mode " + modePencucian + ", " + teksAir;
    }

    public String membilas() {
        return "Sedang membilas " + beratCucian + " kg pakaian, mode " + modePencucian + ", " + teksAir;
    }

    public String mengeringkan() {
        return "Sedang mengeringkan " + beratCucian + " kg pakaian, mode " + modePencucian + ", " + teksAir;
    }

    public String toString() {
        return "Dilakukan oleh mesin cuci dengan merk " + merk + ", tipe " + tipe + ", dengan kapasitas " + kapasitas + " kg";
    }

    public void tampilkan() {
        System.out.println(mencuci());
        System.out.println(membilas());
        System.out.println(mengeringkan());
        System.out.println(toString() + "\n");
    }

    public static void main(String[] args) {
        // Contoh penggunaan dengan input pengguna
        Scanner scanner = new Scanner(System.in);

        System.out.print("Merk: ");
        String merk = scanner.nextLine();
        System.out.print("Tipe (top loading / front loading): ");
        String tipe = scanner.nextLine();
        System.out.print("Kapasitas (kg): ");
        int kapasitas = scanner.nextInt();
        System.out.print("Berat Cucian (kg): ");
        double beratCucian = scanner.nextDouble();
        scanner.nextLine(); 
        System.out.print("Mode Pencucian (ringan / sedang / berat): ");
        String modePencucian = scanner.nextLine();
        System.out.print("Air Hangat (true / false): ");
        boolean airHangat = scanner.nextBoolean();

        MesinCuci mesin = new MesinCuci(merk, tipe, kapasitas, beratCucian, modePencucian, airHangat);

        mesin.tampilkan();

        scanner.close();
    }
}

File: P02 - Kelas Objek/Python/Bag. kedua/MesinCuci.py
# File: MesinCuci.py
# Penulis : Elmosius Suli
# Membuat kelas MesinCuci

class MesinCuci:
    def __init__(self, merk:str, tipe:str, kapasitas:int, beratCucian:float, modePencucian:str, airHangat:bool) -> None:
        self.merk = merk
        self.tipe = tipe if tipe in {"top loading", "front loading"} else "top loading"
        self.kapasitas = kapasitas
        self.beratCucian = beratCucian
        self.modePencucian = modePencucian if modePencucian in {"ringan", "sedang", "berat"} else "sedang"
        self.airHangat = airHangat
        if (airHangat == True):
            self.teksAir = "menggunakan air hangat"
        else:
            self.teksAir = "tidak menggunakan air hangat"
        return

    def mencuci(self) -> str:
        return f"Sedang mencuci {self.beratCucian} kg pakaian, mode {self.modePencucian}, {self.teksAir}"

    def membilas(self) -> str:
        return f"Sedang membilas {self.beratCucian} kg pakaian, mode {self.modePencucian}, {self.teksAir}"

    def mengeringkan(self) -> str:
        return f"Sedang mengeringkan {self.beratCucian} kg pakaian, mode {self.modePencucian}, {self.teksAir}"

    def toString(self) -> str:
        return f"Dilakukan oleh mesin cuci dengan merk {self.merk}, tipe {self.tipe}, dengan kapasitas {self.kapasitas} kg"

    def tampilkan(self) -> None:
        print(self.mencuci())
        print(self.membilas())
        print(self.mengeringkan())
        print(self.toString(), "\n")

    def langkahPencucian(self):
        if (self.beratCucian > self.kapasitas):
            jumlahBagian = self.beratCucian // self.kapasitas + 1
            sisaCucian = self.beratCucian % self.kapasitas
            for i in range (jumlahBagian):
                if (i != jumlahBagian - 1):
                    self.beratCucian = jumlahBagian
                else:
                    self.beratCucian = sisaCucian
                self.tampilkan()
        else:
            self.tampilkan()

## Program utama ##
def main():
    m1 = MesinCuci("Samsung", "front loading", 3, 8, "ringan", True)
    m2 = MesinCuci("LG", "back loading", 6, 2, "sangat berat", False)

    m1.langkahPencucian()
    m2.langkahPencucian()

if __name__ == '__main__':
    main()